galaxy_ng/app/api/ui/serializers/user.py

```diff
@@ -171,68 +171,17 @@ class Meta(UserSerializer.Meta):
-
-            # Collection Distribution
-            "view_distribution": obj.has_perm('ansible.view_ansibledistribution'),
-            "add_distribution": obj.has_perm('ansible.add_ansibledistribution'),
-            "change_distribution": obj.has_perm('ansible.change_ansibledistribution'),
-            "delete_distribution": obj.has_perm('ansible.delete_ansibledistribution'),
-
-            # Container Namespace
-            "add_containernamespace": obj.has_perm('container.add_containernamespace'),
-            "change_containernamespace": obj.has_perm('container.change_containernamespace'),
-            "delete_containernamespace": obj.has_perm('container.delete_containernamespace'),
-
-            # Container Repository
-            "add_containerrepository": obj.has_perm('container.add_containerrepository'),
-            "change_containerrepository": obj.has_perm('container.change_containerrepository'),
-            "delete_containerrepository": obj.has_perm('container.delete_containerrepository'),
-
-            # Container Remote
-            "add_containerregistry": obj.has_perm('galaxy.add_containerregistryremote'),
-            "change_containerregistry": obj.has_perm('galaxy.change_containerregistryremote'),
-            "delete_containerregistry": obj.has_perm('galaxy.delete_containerregistryremote'),
-
-            # Container Distribution
-            "add_containerdistribution": obj.has_perm('container.add_containerdistribution'),
-            "change_containerdistribution": obj.has_perm('container.change_containerdistribution'),
-            "delete_containerdistribution": obj.has_perm('container.delete_containerdistribution'),
-
-            # Tasks
-            "view_task": obj.has_perm('core.view_task'),
-
-            # Auth
-            "view_user": obj.has_perm('galaxy.view_user'),
-            "change_group": obj.has_perm('galaxy.change_group'),
-            "view_group": obj.has_perm('galaxy.view_group'),
-            "delete_user": False,
-            "change_user": False,
-            "add_user": False,
-            "add_group": False,
-            "delete_group": False,
-        }
-
-        if not settings.get("SOCIAL_AUTH_KEYCLOAK_KEY"):
-            permissions["delete_user"] = obj.has_perm('galaxy.delete_user')
-            permissions["change_user"] = obj.has_perm('galaxy.change_user')
-            permissions["add_user"] = obj.has_perm('galaxy.add_user')
-            permissions["add_group"] = obj.has_perm('galaxy.add_group')
-            permissions["delete_group"] = obj.has_perm('galaxy.delete_group')
+        permissions = {}
+        for i, j in PERMISSIONS.items():
+            print(f'i & j: {i, j}')
+            permissions[i] = j
+        for i in permissions:
+            permissions[i]["has_model_permission"] = obj.has_perm(i)
+        if settings.get("SOCIAL_AUTH_KEYCLOAK_KEY"):
+            permissions["galaxy.delete_user"] = False
```

Comment:
This is wrong. You don't want to set `permissions[NAME]` to False. You want to set `permissions[NAME]['has_model_permission']` to False. Remember, this will overwrite the whole permission object, not just the has_model_permission boolean.